lib/msf/core/auxiliary/ubiquiti.rb

```diff
@@ -0,0 +1,336 @@
+# -*- coding: binary -*-
+
+require 'metasploit/framework/hashes/identify'
+require 'bson'
+require 'zip'
+
+module Msf
+
+###
+#
+# This module provides methods for working with Ubiquiti equipment
+#
+###
+module Auxiliary::Ubiquiti
+  include Msf::Auxiliary::Report
+
+  def decrypt_unf(contents)
+    aes = OpenSSL::Cipher.new('aes-128-cbc')
+    aes.key = 'bcyangkmluohmars' # https://github.com/zhangyoufu/unifi-backup-decrypt/blob/master/E>
+    aes.padding = 0
+    aes.decrypt
+    aes.iv = 'ubntenterpriseap'
+    aes.update(contents)
+  end
+
+  def repair_zip(fname)
+    zip_exe = Msf::Util::Helper.which('zip')
+    if zip_exe.nil?
+      return nil
+    end
+    print_status('Attempting to repair zip file (this is normal and takes some time)')
+    temp_file = Rex::Quickfile.new("fixed_zip")
+    system("yes | #{zip_exe} -FF #{fname} --out #{temp_file.path}.zip > /dev/null")
+    if $? == 0
+      return File.read("#{temp_file.path}.zip")
+    else
+      print_error('Error fixing zip.  Attempt manually.')
+      nil
```

Comment:
The `nil` here isn't necessary. The whole `else` block could actually be removed since the conditional block above it always returns.